src/routing_node.rs

```diff
@@ -836,6 +842,64 @@ impl RoutingNode {
                                   refresh_our_own_group: bool) -> RoutingResult {
         unimplemented!()
     }
+
+    fn handle_cache_get(&mut self, request : ExternalRequest,
+                                   from_authority : Authority, response_token : SignedToken)
+         -> RoutingResult {
```

Comment:
I wonder if this is not more complex than needed. It could return an Option<Data> so if the result is `Some(data)` we could call the reply method with the data and appropriate info. Then this is just a fn called when there is a get_data_request message.